Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts

```diff
@@ -134,18 +96,30 @@ async function doWork() {
         }
         tl.debug(tl.loc('GetArtifactsFromBuildNumber', buildId, jobName));
 
-        const artifactsRelativePaths: string[] = await getArtifactsRelativePaths(serverEndpointUrl, jobName, buildId, cred, strictSSL);
-        artifactsRelativePaths.forEach((relativePath) => {
-            const localPath: string = path.resolve(localPathRoot, relativePath);
-            const dir: string = path.dirname(localPath);
-            if (!tl.exist(dir)) {
-                tl.mkdirP(dir);
-            }
-
-            const artifactUrl: string = getArtifactUrl(serverEndpointUrl, jobName, buildId, relativePath);
-            //Q: Are we supposed to await on this download call?
-            download(artifactUrl, localPath, cred, strictSSL);
-        });
+        const artifactQueryUrl: string = `${serverEndpointUrl}/job/${jobName}/${buildId}/api/json?tree=artifacts[*]`;
+        console.log(tl.loc('ArtifactDownloadUrl', artifactQueryUrl));
+
+        var templatePath = path.join(__dirname, 'jenkins.handlebars.txt');
+        var variables = {
+            "endpoint": {
+                "url": serverEndpointUrl
+            },
+            "definition": jobName,
+            "version": buildId
+        };
+        var handler = new handlers.BasicCredentialHandler(username, password);
+        
+        var webProvider = new providers.WebProvider(artifactQueryUrl, templatePath, variables, handler, { ignoreSslError: strictSSL });
+        var localFileProvider = new providers.FilesystemProvider(localPathRoot);
+    
+        let downloader = new engine.ArtifactEngine();
+        var downloaderOptions = new engine.ArtifactEngineOptions();
+        downloaderOptions.itemPattern = tl.getInput('itemPattern', false) || "**";
+        downloaderOptions.parallelProcessingLimit = +tl.getVariable("release.artifact.download.parallellimit") || 4;
```

Comment:
the + at the start is to cast to integer?

Reply:
yes